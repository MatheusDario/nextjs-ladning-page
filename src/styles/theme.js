export const theme = {
  colors: {
    white: '#f0f0f2',
    blue: '#023059',
    lightblue: '#065473',
    aquablue: '#08748c',
    green: '#02735e',
    lightgreen: '#6ab04c',
    lightgray: ' #cecece',
    red: '#dc143c',
    black: '#000000',
  },
  spacing: {
    xmall: '.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
  },
  font: {
    sizes: {
      xmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
    },
    family: {
      default: "'Lato', sans-serif",
      title: "'Montserrat', sans-serif",
    },
  },
  breakpoints: {
    mobile: '@media (max-width:768px)',
    tests: '(max-width:768px)',
  },
};
