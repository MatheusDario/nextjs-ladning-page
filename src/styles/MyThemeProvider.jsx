import { ThemeProvider } from "styled-components";
import { theme } from "./theme";
import React from "react";
import { GlobalStyles } from "./global-styles";

export default function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}

// MyThemeProvider.propTypes = {
//   children: P.node.isRequired,
// };
