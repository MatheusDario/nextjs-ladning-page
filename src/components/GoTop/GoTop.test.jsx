import { screen } from '@testing-library/react';
import { GoTop } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<GoTop />', () => {
  it('should render a go to top button', () => {
    const { container } = renderTheme(<GoTop />);
    expect(
      screen.getByRole('link', { name: 'Ir para o inicio' }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: 'Ir para o inicio' }),
    ).toHaveAttribute('href', '#');
    expect(container).toMatchSnapshot();
  });
});
