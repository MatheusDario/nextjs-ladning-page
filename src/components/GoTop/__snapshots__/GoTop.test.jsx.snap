// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoTop /> should render a go to top button 1`] = `
.c0 {
  background-color: #02735e;
  color: #f0f0f2;
  border-radius: 50%;
  opacity: .7;
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  right: 2rem;
  bottom: 2rem;
  z-index: 6;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<div>
  <a
    aria-label="Ir para o inicio"
    class="c0"
    href="#"
    title="Voltar para o comeÃ§o"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
